client/state/sites/plans/actions.js

```diff
@@ -31,39 +28,13 @@ import wpcom from 'lib/wp';
  * @returns {Function} a promise that will resolve once updating is completed
  */
 export function cancelSitePlanTrial( siteId, planId ) {
-	return ( dispatch ) => {
-		dispatch( {
-			type: SITE_PLANS_TRIAL_CANCEL,
-			siteId
-		} );
-
-		return new Promise( ( resolve, reject ) => {
-			wpcom.undocumented().cancelPlanTrial( planId, ( error, data ) => {
-				if ( data && data.success ) {
-					dispatch( {
-						type: SITE_PLANS_TRIAL_CANCEL_COMPLETED,
-						siteId,
-						plans: map( data.plans, createSitePlanObject )
-					} );
-
-					resolve();
-				} else {
-					debug( 'Canceling site plan trial failed: ', error );
-
-					const errorMessage = error.message || i18n.translate( 'There was a problem canceling the plan trial. Please try again later or contact support.' );
-
-					dispatch( {
-						type: SITE_PLANS_TRIAL_CANCEL_FAILED,
-						siteId,
-						error: errorMessage
-					} );
-
-					reject( errorMessage );
-				}
-			} );
-		} );
+	return {
+		type: SITE_PLANS_TRIAL_CANCEL,
+		siteId,
+		planId
 	}
 }
+registerConnection( SITE_PLANS_TRIAL_CANCEL, connectCancelSitePlanTrial, { offlineQueue: false } );
```

Comment:
> curious... why wouldn't you want to replay cancelling a trial?

I wanted to clearly illustrate, that this approach is not volotile and you can completely bypass whole queuing system and then it will actually be executed the same way as now.

> I think the call to registerConnection here would be better placed within connections.js

I was actually considering that for a while and I did it this way for 2 reasons:
- If everything is encapsulated in `connections.js` then where do we require the file and run proper functions?
- when the connections are registered close to actions that trigger them, its easy to see what happens next. Otherwise it would be less obvious to track down the actual WPCOM request that comes out. This way, we at least have an indication that there is one.